vllm/model_executor/model_loader/weight_utils.py

```diff
@@ -514,8 +515,29 @@ def convert_pyslice_to_tensor(x: Any) -> torch.Tensor:
 def default_weight_loader(param: torch.Tensor,
                           loaded_weight: torch.Tensor) -> None:
     """Default weight loader."""
-    assert param.size() == loaded_weight.size()
-    param.data.copy_(loaded_weight)
+    try:
+        assert param.size() == loaded_weight.size(), (
+            f"Attempted to load weight ({loaded_weight.size()}) "
+            f"into parameter ({param.size()})")
+
+        param.data.copy_(loaded_weight)
+    except Exception:
+        # You can set a breakpoint here for debugging weight loading issues
+        raise
```

Comment:
Let's make this comment a note and more formal.
```suggestion
        # NOTE: This exception is added for the purpose of setting breakpoint to 
        # debug weight loading issues. 
        raise
```